### DISCUSSION TITLE
怎么实现设计时绑定

### DISCUSSION THREAD

**Discussion by notplay3** [BYJRK/DotNet-Discussions#53] [WPF 开发]
我想使用设计时绑定,在没有运行的时候就看到绑定后ui显示后台数据的值,百度后
d:DataContext="{d:DesignInstance vm:MainViewModel}"使用这个,但是使用这个只能在binding时可以智能提示有哪些属性,并不能将后台数据反应到UI上,求解?,谢谢!

**Comment by BYJRK** [comment]
如果那些属性的值是编译时就可以得到的，那么他们是可以直接显示的。否则，如果是运行时才得到的，是无法显示的。你可以给属性赋初始值，或者可以用 `Binding` 的 `FallbackValue` 属性，还可以用诸如 `d:Text` 等属性来实现设计时的假数据。
